include/net/net_if.h

```diff
@@ -1385,6 +1385,32 @@ bool net_if_ipv4_addr_mask_cmp(struct net_if *iface,
  */
 struct net_if *net_if_ipv4_select_src_iface(struct in_addr *dst);
 
+/**
+ * @brief Get a IPv4 source address that should be used when sending
+ * network data to destination.
+ *
+ * @param iface Interface that was used when packet was received.
```

Comment:
It's a bit strange to have the description "Interface that was used when packet was received", while the name for this parameter - `dst_iface`.